### DISCUSSION TITLE
Exclude formatting on save for certain languages

### DISCUSSION THREAD

**Discussion by JamesTeague** [VonHeikemen/lsp-zero.nvim#356] [Q&A]
I have lsp-zero.nvim and ray-x/go.nvim as plugins in my setup. When writing files, the formatting sometimes get all messed up. Examples include losing the first character of every line in the file, or dropping necessary double quotes. I believe it is because I have 2 plugins trying to format the file. Is it possible to tell lsp-zero to format everything except Go files?

**Comment by JamesTeague** [comment]
This should be the solution [here](https://github.com/ray-x/go.nvim/issues/131) for my specific case. But I would still like to know the answer to the original question. I think that would be useful anyway.

**Comment by VonHeikemen** [comment]
How did you setup format on save?

**Reply by JamesTeague** [reply]
I don't think I have explicitly.

```lua
local lsp = require('lsp-zero').preset("recommended")
-- cmp mappings and some on_attach things
... 

lsp.setup()
```
I've truncated what I thought was not import but that is all I have right now.

**Reply by VonHeikemen** [reply]
That doesn't enable format on save.

Do you use any of these functions?

* [.buffer_autoformat()](https://github.com/VonHeikemen/lsp-zero.nvim/blob/v3.x/doc/md/api-reference.md#buffer_autoformatclient-bufnr-opts) 
* [.async_autoformat()](https://github.com/VonHeikemen/lsp-zero.nvim/blob/v3.x/doc/md/api-reference.md#async_autoformatclient-bufnr-opts)
* [.format_on_save()](https://github.com/VonHeikemen/lsp-zero.nvim/blob/v3.x/doc/md/api-reference.md#format_on_saveopts)
